### PR TITLE
Run the flex container tests against UIViewLazyList

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: redwood-layout-uiview/src/commonTest/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainerTest.kt [hunk 1]
```diff
@@ -41,10 +41,6 @@ class UIViewFlexContainerTest(
     }
   }
 
-  override fun row() = flexContainer(FlexDirection.Row)
-
-  override fun column() = flexContainer(FlexDirection.Column)
```
Comment (swankjesse) [inline redwood-layout-uiview/src/commonTest/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainerTest.kt:46]: These were nearly unused.

File: redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt [hunk 2]
```diff
@@ -592,18 +586,19 @@ abstract class AbstractFlexContainerTest<T : Any> {
...
     container.children.insert(0, text(mediumText(), MarginImpl(10.dp), Green))
     container.children.insert(1, text(shortText(), MarginImpl(0.dp), Blue))
     container.onEndChanges()
-    verifySnapshot(container, "Before")
+    verifySnapshot(widget, "Before")
 
     // Detach after changes are applied but before they're rendered.
     container.children.insert(0, text(longText(), MarginImpl(20.dp), Red))
     container.onEndChanges()
     container.children.detach()
```
Comment (swankjesse) [inline redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt:606]: This is what I wanted to borrow

File: redwood-lazylayout-uiview/build.gradle [hunk 3]
```diff
@@ -25,6 +25,8 @@ kotlin {
     }
     commonTest {
       dependencies {
+        api projects.redwoodLayoutSharedTest
+        api projects.redwoodLayoutUiview
```
Comment (swankjesse) [inline redwood-lazylayout-uiview/build.gradle:29]: Unfortunately one of the tests currently assumes `Column`, not an arbitrary container.

File: redwood-lazylayout-uiview/src/commonTest/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyListAsFlexContainerTest.kt [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  override fun flexContainer(
+    direction: FlexDirection,
+    backgroundColor: Int,
+  ) = ViewTestFlexContainer(widgetFactory.LazyList(), direction, backgroundColor)
+
+  override fun row() = UIViewRedwoodLayoutWidgetFactory().Row()
+
+  override fun column() = UIViewRedwoodLayoutWidgetFactory().Column()
+
+  override fun text() = object : Text<UIView> {
```
Comment (JakeWharton) [inline redwood-lazylayout-uiview/src/commonTest/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyListAsFlexContainerTest.kt:51]: #2275

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**